client/api_challenge.go

```diff
@@ -37,8 +38,8 @@ func (c *client) GetChallengeInfo(ctx context.Context, info types.ChallengeInfo)
 		return types.ChallengeResult{}, errors.New("index error, should be 0 to parityShards plus dataShards")
 	}
 
-	if info.RedundancyIndex < -1 {
-		return types.ChallengeResult{}, errors.New("redundancy index error ")
+	if info.RedundancyIndex < -1 || info.RedundancyIndex > 5 {
```

Comment:
DO NOT hard code here. It's best to query the on-chain parameters of redundancy.

Reply:
updated